src/validators/validateBin.test.ts

```diff
@@ -0,0 +1,92 @@
+import { describe, expect, it } from "vitest";
+
+import { validateBin } from "./validateBin";
+
+describe("validateBin", () => {
+	it("should return no errors if the bin field is an empty object", () => {
+		const result = validateBin({});
+		expect(result).toEqual([]);
+	});
+
+	it("should return no errors if the bin field is a valid string", () => {
+		let result = validateBin("./cli.js");
+		expect(result).toEqual([]);
+
+		result = validateBin("cli.js");
+		expect(result).toEqual([]);
+
+		result = validateBin("./bin/cli.js");
+		expect(result).toEqual([]);
+
+		result = validateBin("bin/cli.js");
+		expect(result).toEqual([]);
+	});
```

Comment:
[Style] Nit: for more granular cases, and a bit of LoC reduction, how about a `.each`?

```suggestion
	it.each([
		"./cli.js", "cli.js", "./bin/cli.js", "bin/cli.js",
	])("should return no errors if the bin field is a valid string: %s", (binPath) => {
		const result = validateBin(binPath);
		expect(result).toEqual([]);
	});
```